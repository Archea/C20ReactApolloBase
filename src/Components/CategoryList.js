import React, { Component } from 'react'
import CategoryGrid from './CategoryGrid'
import { graphql, Query } from 'react-apollo'
import gql from 'graphql-tag'

export class CategoryList extends Component {
  render() {
    if (this.props.getState && this.props.getState.loading) {
      return <div>Loading</div>
    }

    if (this.props.getState && this.props.getState.error) {
      console.log(
        'Get Categories state error: ' + this.props.getState.error.message
      )
      return <div>Error</div>
    }
    const firmId = this.props.getState.userdata.firm.id
    //When we need teh activeCategory can use below (if not null...)
    return (
      <div className="CategoryList">
        <Query query={CATAGORY_QUERY} variables={{ firmId }}>
          {({ loading, error, data, client }) => {
            if (error) return `Error! ${error.message}`
            if (loading) return 'loading...'
            return <CategoryGrid {...data} />
          }}
        </Query>
      </div>
    )
  }
}

/*
This query gets all the ProductFamilies aka sections from content20 db
the 'name' is the section title, while the id is the mf section#
*/
const CATAGORY_QUERY = gql`
  query GetAllPFC($firmId: String!) {
    allProductFamilyCategories(firmId: $firmId) {
      id
      name
      discipline
      firm {
        name
      }
      selected @client
    }
  }
`
/*
  firm id is needed for the queries
  we pull it from the local userdata
  on load with the HoC
*/
const INITIALSTATE_QUERY = gql`
  query GetInitSate {
    userdata @client {
      firm {
        id
      }
    }
  }
`
export default graphql(INITIALSTATE_QUERY, { name: 'getState' })(CategoryList)
