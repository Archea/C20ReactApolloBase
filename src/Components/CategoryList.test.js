import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, configure } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { CategoryList } from './CategoryList'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe('CategoryList', () => {
  const mocks = [
    {
      id: '1',
      name: 'bar'
    },
    {
      id: '2',
      name: 'foosle'
    }
  ]
  const mockGetAll = jest.fn
  const mockprops = {
    getState: {
      loading: false,
      error: null,
      userdata: {
        firm: {
          id: '0'
        }
      }
    }
  }
  test('it at least shallow renders and match snapshot', () => {
    const wrapper = shallow(<CategoryList {...mockprops} />)
    wrapper.setProps(mockprops)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
  const mockpropsError = {
    getState: {
      loading: false,
      error: {
        message: 'an error was mocked in your system'
      }
    }
  }
  test('it handles a query error', () => {
    const wrapper = shallow(<CategoryList {...mockpropsError} />)
    wrapper.setProps(mockpropsError)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
  const mockpropsLoading = {
    getState: {
      loading: true,
      error: null
    }
  }
  test('it handles a query loading', () => {
    const wrapper = shallow(<CategoryList {...mockpropsLoading} />)
    wrapper.setProps(mockpropsLoading)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
