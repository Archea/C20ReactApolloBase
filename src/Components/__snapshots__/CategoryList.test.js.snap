// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList it at least shallow renders and match snapshot 1`] = `
<div
  className="CategoryList"
>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "GetAllPFC",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "firmId",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "firmId",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "allProductFamilyCategories",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "discipline",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "firm",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [
                          Object {
                            "arguments": Array [],
                            "kind": "Directive",
                            "name": Object {
                              "kind": "Name",
                              "value": "client",
                            },
                          },
                        ],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "selected",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "firmId",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 193,
          "start": 0,
        },
      }
    }
    variables={
      Object {
        "firmId": "0",
      }
    }
  />
</div>
`;

exports[`CategoryList it handles a query error 1`] = `
<div>
  Error
</div>
`;

exports[`CategoryList it handles a query loading 1`] = `
<div>
  Loading
</div>
`;
